modules/tensor_mechanics/src/materials/LAROMANCEPartitionStressUpdateBase.C

```diff
@@ -187,7 +194,8 @@ void
 LAROMANCEPartitionStressUpdateBaseTempl<is_ad>::computeDSecondPartitionWeightDStress(
     GenericReal<is_ad> & dsecond_partition_weight_dstress)
 {
-  const Real fd_tol = 1.0e-6; // Finite difference width
+  const auto fd_tol =
+      _input_values[_stress_input_index] * _finite_difference_width; // Finite difference width
```

Comment:
Here is where the finite difference width becomes relative @arovinelli , e.g. relative to the input stress, which is what we are trying to estimate the derivative for. The relative value could be important for high stresses, where the absolute 1e-6 may get washed out in numerical precision. @aruybalid is looking into creating an analytical derivative for this as well, so hopefully we can wipe this out...

Reply:
if an analytic derivative is already planned then I don't really mind the FD value. Still, I'm wondering if we might want to make `fd_tol` an adjustable parameter with a default value that we consider as optimal.

Reply:
Well, `_finite_difference_width` is an input parameter...is that what you mean, or you still think it should be absolute?

Reply:
Sorry I got confused from reading it from email I'm with `_finite_difference_width` being relative, and I'm fine with it being an optional input parameter !

Reply:
Thanks for taking a look @arovinelli !

Reply:
FYI @lynnmunday here's the fix that will help you (I hope!) until this gets merged.